contracts/BSCValidatorSet.sol

```diff
@@ -478,6 +526,26 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
       uint256 newMaintainSlashScale = BytesToTypes.bytesToUint256(32, value);
       require(newMaintainSlashScale > 0, "the maintainSlashScale must be greater than 0");
       maintainSlashScale = newMaintainSlashScale;
+    } else if (Memory.compareStrings(key, "maxNumOfWorkingCandidates")) {
+      require(value.length == 32, "length of maxNumOfWorkingCandidates mismatch");
+      uint256 newMaxNumOfWorkingCandidates = BytesToTypes.bytesToUint256(32, value);
+      require(newMaxNumOfWorkingCandidates > 0, "the maxNumOfWorkingCandidates must be not less than 0");
+      uint256 _maxNumOfCandidates = maxNumOfCandidates;
+      if (_maxNumOfCandidates == 0) {
+        _maxNumOfCandidates = INIT_MAX_NUM_OF_CANDIDATES;
+      }
+      require(newMaxNumOfWorkingCandidates <= _maxNumOfCandidates, "the maxNumOfCandidates must be not greater than maxNumOfCandidates");
+      maxNumOfWorkingCandidates = newMaxNumOfWorkingCandidates;
+    } else if (Memory.compareStrings(key, "maxNumOfCandidates")) {
+      require(value.length == 32, "length of maxNumOfCandidates mismatch");
+      uint256 newMaxNumOfCandidates = BytesToTypes.bytesToUint256(32, value);
+      require(newMaxNumOfCandidates > 0, "the maxNumOfCandidates must be not less than 0");
+      maxNumOfCandidates = newMaxNumOfCandidates;
```

Comment:
If we follow the same code stye, you should init maxNumOfCandidates it the value is zero. But I suggest delete the init of  _maxNumOfCandidates too.

Reply:
updated

Reply:
In the initial state,`maxNumOfCandidates ` and `maxNumOfWorkingCandidates ` will be 0, which means that candidate validators will not join the consensus